gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspConfigurations.kt

```diff
@@ -109,68 +110,125 @@ class KspConfigurations(private val project: Project) {
     }
 
     /**
-     * Decorate the [KotlinSourceSet]s belonging to [target] to create one KSP configuration per source set,
-     * named ksp<SourceSet>. The only exception is the main source set, for which we avoid using the
-     * "main" suffix (so what would be "kspJvmMain" becomes "kspJvm").
+     * Decorate [target]'s source sets (Android) or compilations (non-Android), creating one KSP configuration
+     * per source set or compilation.
      *
      * For Android, we prefer to use AndroidSourceSets from AGP rather than [KotlinSourceSet]s.
      * Even though the Kotlin Plugin does create [KotlinSourceSet]s out of AndroidSourceSets
      * ( https://kotlinlang.org/docs/mpp-configure-compilations.html#compilation-of-the-source-set-hierarchy ),
      * there are slight differences between the two - Kotlin creates some extra sets with unexpected word ordering,
      * and things get worse when you add product flavors. So, we use AGP sets as the source of truth.
+     * Android configurations are named ksp<SourceSet>, stripping a "Main" suffix (so what would be "kspJvmMain"
+     * becomes "kspJvm").
+     *
+     * Non-Android compilations are named ksp<Target><Compilation> except for main compilations, which are
+     * named ksp<Target>.
      */
     private fun decorateKotlinTarget(target: KotlinTarget) {
+        // TODO: Check whether special AGP handling is still necessary.
```

Comment:
The motivation for this comment is this: Previously, KSP could only see a subset of source sets configured. This has now changed, so it might see AGP-created source sets where it did not before, possibly enabling a unified KMP/Android handling. Would you suggest dropping this comment without further investigation?